src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -912,6 +931,25 @@ MulticopterPositionControl::reset_alt_sp()
 	}
 }
 
+void
+MulticopterPositionControl::limit_altitude()
+{
+	float alt = _pos(2);
+
+	/* in altitude control, limit setpoint */
+	if (-alt >= _valid_altitude_max && _run_alt_control && _pos_sp(2) <= _pos(2)) {
```

Comment:
why not choose `_valid_altitude_max` in the NED frame? then it would be the parameter negative and consistent with alt and _pos_sp(2)